node/src/database/db_migrations/db_migrator.rs

```diff
@@ -0,0 +1,808 @@
+            self
+        }
+
+        pub fn execute_upon_transaction_result(self, result: rusqlite::Result<()>) -> Self {
+            self.execute_upon_transaction_results
+                .borrow_mut()
+                .push(result);
+            self
+        }
+    }
+
+    impl MigDeclarationUtilities for DBMigrateDeclarationUtilitiesMock {
+        fn db_password(&self) -> Option<String> {
+            self.db_password_results.borrow_mut().remove(0)
+        }
+
+        fn transaction(&self) -> &Transaction {
+            unimplemented!("Not needed so far")
+        }
+
+        fn execute_upon_transaction<'a>(
+            &self,
+            sql_statements: &[&'a dyn StatementObject],
+        ) -> rusqlite::Result<()> {
+            self.execute_upon_transaction_params.lock().unwrap().push(
+                sql_statements
+                    .iter()
+                    .map(|stm_obj| stm_obj.to_string())
+                    .collect::<Vec<String>>(),
+            );
+            self.execute_upon_transaction_results.borrow_mut().remove(0)
+        }
+
+        fn external_parameters(&self) -> &ExternalData {
+            unimplemented!("Not needed so far")
+        }
+
+        fn logger(&self) -> &Logger {
+            unimplemented!("Not needed so far")
+        }
+    }
+
+    #[derive(Default, Debug)]
+    struct DBMigrationRecordMock {
+        old_version_result: RefCell<usize>,
+        migrate_params: Arc<Mutex<Vec<()>>>,
+        migrate_result: RefCell<Vec<rusqlite::Result<()>>>,
+    }
+
+    impl DBMigrationRecordMock {
+        fn old_version_result(self, result: usize) -> Self {
+            self.old_version_result.replace(result);
+            self
+        }
+
+        fn migrate_result(self, result: rusqlite::Result<()>) -> Self {
+            self.migrate_result.borrow_mut().push(result);
+            self
+        }
+
+        fn migrate_params(mut self, params: &Arc<Mutex<Vec<()>>>) -> Self {
+            self.migrate_params = params.clone();
+            self
+        }
+
+        fn set_up_necessary_stuff_for_mocked_migration_record(
+            self,
+            result_o_v: usize,
+            result_m: rusqlite::Result<()>,
+            params_m: &Arc<Mutex<Vec<()>>>,
+        ) -> Self {
+            self.old_version_result(result_o_v)
+                .migrate_result(result_m)
+                .migrate_params(params_m)
+        }
+    }
+
+    impl DatabaseMigration for DBMigrationRecordMock {
+        fn migrate<'a>(
+            &self,
+            _migration_utilities: Box<dyn MigDeclarationUtilities + 'a>,
+        ) -> rusqlite::Result<()> {
+            self.migrate_params.lock().unwrap().push(());
+            self.migrate_result.borrow_mut().remove(0)
+        }
+
+        fn old_version(&self) -> usize {
+            *self.old_version_result.borrow()
+        }
+    }
+
+    const _REMINDER_FROM_COMPILATION_TIME: () = check_schema_version_continuity();
+
+    #[allow(dead_code)]
```

Comment:
I like the idea of a compile-time test like this, but the necessity for this directive makes me suspicious that the compiler might be optimizing away both `_REMINDER_FROM_COMPILATION_TIME` and `check_schema_version_continuity()`.